apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -99,8 +99,30 @@ type ClusterResourcePlacementSpec struct {
 	// +kubebuilder:default=10
 	// +optional
 	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty"`
+
+	// ApplyMode defines how to apply resources if the resources exist on the target clusters.
+	// It can be "CreateOnly", "ServerSideApply" or "ServerSideApplyWithForceConflicts". Default is CreateOnly.
+	// +kubebuilder:validation:Enum=CreateOnly;ServerSideApply;ServerSideApplyWithForceConflicts
+	// +kubebuilder:default=CreateOnly
+	// +optional
+	ApplyMode ApplyMode `json:"applyMode,omitempty"`
 }
 
+// ApplyMode defines how resources are applied on the target clusters.
+type ApplyMode string
+
+const (
+	// ApplyModeCreateOnly creates the resources on the target clusters and will fail if the resources exist on the target
+	// cluster.
+	ApplyModeCreateOnly ApplyMode = "CreateOnly"
+	// ApplyModeServerSideApply applies the resources on the target cluster using server-side apply.
+	// https://kubernetes.io/docs/reference/using-api/server-side-apply/
+	ApplyModeServerSideApply ApplyMode = "ServerSideApply"
+	// ApplyModeServerSideApplyWithForceConflicts applies the resources on the target cluster using server-side apply with
+	// force-conflicts option. It forces the operation to succeed when resolving the conflicts.
+	ApplyModeServerSideApplyWithForceConflicts ApplyMode = "ServerSideApplyWithForceConflicts"
+)
```

Comment:
this is essentially expressing two different controls
1. server side apply or not
2. force apply or not
I wonder if it would make more sense to expand this single mode into a struct so we can add more options in the future?